src/OpenCensus.Collector.StackExchangeRedis/Implementation/RedisProfilerEntryToSpanConverter.cs

```diff
@@ -0,0 +1,125 @@
+            // command.ResponseToCompletion;    // 00:00:00.0002601
+
+            // Total:
+            // command.ElapsedTime;             // 00:00:32.4988020
+
+            // TODO: make timestamp with the better precision
+            ITimestamp startTimestamp = Timestamp.FromMillis(new DateTimeOffset(command.CommandCreated).ToUnixTimeMilliseconds());
+
+            var timestamp = new DateTimeOffset(command.CommandCreated).Add(command.CreationToEnqueued);
+            var annotations = TimedEvents<IAnnotation>.Create(
+                new List<ITimedEvent<IAnnotation>>()
+                {
+                    TimedEvent<IAnnotation>.Create(Timestamp.FromMillis(timestamp.ToUnixTimeMilliseconds()), Annotation.FromDescription("Enqueued")),
+                    TimedEvent<IAnnotation>.Create(Timestamp.FromMillis((timestamp = timestamp.Add(command.EnqueuedToSending)).ToUnixTimeMilliseconds()), Annotation.FromDescription("Sent")),
+                    TimedEvent<IAnnotation>.Create(Timestamp.FromMillis((timestamp = timestamp.Add(command.SentToResponse)).ToUnixTimeMilliseconds()), Annotation.FromDescription("ResponseRecieved")),
+                },
+                droppedEventsCount: 0);
+
+            ITimestamp endTimestamp = Timestamp.FromMillis(new DateTimeOffset(command.CommandCreated.Add(command.ElapsedTime)).ToUnixTimeMilliseconds());
+
+            // TODO: deal with the re-transmission
+            // command.RetransmissionOf;
+            // command.RetransmissionReason;
+
+            var attributes = Attributes.Create(
+                new Dictionary<string, IAttributeValue>()
+                {
+                    // Example: "db.statement": SET;
+                    { "db.statement", AttributeValue.StringAttributeValue(command.Command) },
+
+                    // TODO: pre-allocate constant attribute and reuse
+                    { "db.type", AttributeValue.StringAttributeValue("redis") },
+
+                    // Example: "db.instance": Unspecified/localhost:6379[0]
+                    { "db.instance", AttributeValue.StringAttributeValue(command.EndPoint.ToString() + "[" + command.Db + "]") },
+
+                    // Example: "redis.flags": None, DemandMaster
+                    { "redis.flags", AttributeValue.StringAttributeValue(command.Flags.ToString()) },
+                },
+                0);
+
+            ITimedEvents<IMessageEvent> messageOrNetworkEvents = null;
+            ILinks links = null;
+            int? childSpanCount = 0;
+
+            // TODO: this is strange that IProfiledCommand doesn't give the result
+            Status status = Status.Ok;
+            SpanKind kind = SpanKind.Client;
+
+            // TODO: SpanData.Create is from OpenCensus implementation
+            return SpanData.Create(context, parentSpanId, hasRemoteParent, name, startTimestamp, attributes, annotations, messageOrNetworkEvents, links, childSpanCount, status, kind, endTimestamp);
```

Comment:
as per offline discussion, I suggest that we create an ISpan as usual (via Tracer) and use InternalsVisibleTo or private reflection to set start time in the past.